From cef0f4b3bc252254b45e0bc81594e88a104ccb7f Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 16 Oct 2023 10:29:54 +0100
Subject: [PATCH 1/1] Add -Wl,-z,pack-relative-relocs to default linker flags

We use both PIE and symbolic-functions so we're ripe for benefitting from
this for smaller binaries and potentially better cache locality.
---
 pisi/configfile.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index 4bba92e..60c49f7 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -86,7 +86,7 @@ class BuildDefaults:
     enableSandbox = False # Dropping sandbox support soon
     cflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -fno-plt -Wformat -Wformat-security -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wall -Wno-error -Wp,-D_REENTRANT"
     cxxflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -fno-plt -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wall -Wno-error -Wp,-D_REENTRANT"
-    ldflags = "-Wl,--copy-dt-needed-entries -Wl,-O1 -Wl,-z,relro -Wl,-z,now -Wl,-z,max-page-size=0x1000 -Wl,-Bsymbolic-functions -Wl,--sort-common"
+    ldflags = "-Wl,--copy-dt-needed-entries -Wl,-O1 -Wl,-z,relro -Wl,-z,now -Wl,-z,max-page-size=0x1000 -Wl,-Bsymbolic-functions -Wl,--sort-common -Wl,-z,pack-relative-relocs"
     buildhelper = "ccache"
     compressionlevel = 1
     fallback = "https://sources.getsol.us/"
-- 
2.42.0

